src/ConfigFactory.php

```diff
@@ -0,0 +1,228 @@
+            );
+        }
+
+        $fs = self::getFilesystem();
+
+        if (!$fs->exists($home)) {
+            $fs->mkdir($home, 0744);
+        }
+
+        return $home;
+    }
+
+    /**
+     * Create a new Config object with the system and (optional) local configuration
+     * loaded.
+     *
+     * Note that any missing config file is ignored.
+     *
+     * @param string|null $localHome Local home folder to load extra configuration from
+     *                               when null this is ignored
+     *
+     * @return Config
+     */
+    public static function createConfig($localHome = null)
+    {
+        $home = static::getHomedir();
+        $cacheDir = self::getCacheDir($home, $home);
+        $localConfig = [];
+
+        $systemConfig = self::loadFileOrEmpty($home.'/.gush.yml');
+
+        // Remove old unsupported configuration format
+        // this forces a clean config structure
+        if (isset($systemConfig['parameters']) && 1 === count($systemConfig)) {
+            $systemConfig = [];
+        }
+
+        if (null !== $localHome) {
+            $localConfig = self::loadFileOrEmpty($localHome.'/.gush.yml');
+        }
+
+        $config = new Config($home, $cacheDir, $systemConfig, $localHome, $localConfig);
+
+        $fs = self::getFilesystem();
+
+        // Protect directory against web access. Since HOME could be
+        // the www-data's user home and be web-accessible it is a
+        // potential security risk
+        foreach ([$home, $cacheDir] as $dir) {
+            if (!file_exists($dir.'/.htaccess')) {
+                $fs->mkdir($dir);
+
+                file_put_contents($dir.'/.htaccess', 'Deny from all');
+            }
+        }
+
+        return $config;
+    }
+
+    /**
+     * Dump configuration to the related config-file.
+     *
+     * The config-file is automatically determined using
+     * type.
+     *
+     * @param Config $config
+     * @param string $type
+     *
+     * @return string
+     */
+    public static function dumpToFile(Config $config, $type)
+    {
+        if ($type === Config::CONFIG_LOCAL) {
+            $filename = $config->get('local_config');
+
+            if (null === $filename) {
+                throw new \RuntimeException('Local configuration is not loaded and therefor can not be dumped.');
+            }
+        } elseif ($type === Config::CONFIG_SYSTEM) {
+            $filename = $config->get('home_config');
+        } else {
+            throw new \InvalidArgumentException(
+                sprintf(
+                    'Config slot "%s" is not valid for dumping to a file "%s", use either: '.
+                    'Config::CONFIG_SYSTEM or Config::CONFIG_LOCAL.',
+                    $type
+                )
+            );
+        }
+
+        // Testing compatibility, write directly as there is no risk of problems
+        if ('vfs://' === substr($filename, 0, 6)) {
```

Comment:
The Symfony `Filesystem::dump()` method uses tempnam(), and you can't move/copy from one filesystem to another without using stream_copy().

Setting the TMP folder at runtime is not possible (security) and setting it before running may cause some unexpected problems with PHPUnit runInSeparateProcess.

Normally I don't like hacks like this but its a small one and only at one place.